php/commands/core.php

```diff
@@ -134,7 +134,8 @@ public function download( $args, $assoc_args ) {
 		$locale = \WP_CLI\Utils\get_flag_value( $assoc_args, 'locale', 'en_US' );
 
 		if ( isset( $assoc_args['version'] ) ) {
-			$version = $assoc_args['version'];
+			$version = strtolower( $assoc_args['version'] );
+			$version = ( 'nightly' === $version ? 'trunk' : $version );
 			$download_url = $this->get_download_url($version, $locale, 'tar.gz');
```

Comment:
> Potentially yes. What would be the best way of handling this? We should ask for what we actually want.

Given it's a private method, so we don't need to worry about backwards compat concerns, could we just remove the third argument and make sure the calling function handles the archive file correctly, regardless of type?

> But what should happen if we ask for a 'tar.gz' of the nightly build - an exception?

Generally, we aren't using exceptions within WP-CLI. Calling `WP_CLI::error()` is a better way of handling errors where WP-CLI doesn't know how to proceed.